src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -161,28 +177,51 @@ private boolean match(VendorData vendorData) {
   /**
    * Checks if product data from an NVD entry matches to the project's name.
    *
-   * @param productData The produce data.
+   * @param productData The product data.
+   * @param project of type {@link GitHubProject}.
    * @return True if the product data matches to the project's name, false otherwise.
    */
-  private boolean match(ProductData productData) {
+  private static boolean match(ProductData productData, GitHubProject project) {
     return match(project.name(), productData.getProductName());
   }
 
   /**
-   * Checks if a string is similar to another. It goes through series of controlled validations to
+   * Checks if a string is similar to another. It goes through a series of controlled validations to
    * confirm this.
    * 
    * @param String one.
    * @param String two.
    * @return True if the validations part of this method is satisfied. Otherwise False.
    */
-  private boolean match(String one, String two) {
+  private static boolean match(String one, String two) {
     if (one.equalsIgnoreCase(two)) {
       return true;
     }
 
     return jaroWinklerSimilarityCheck(one, two) && longestCommonSubsequenceCheck(one, two);
   }
+  
+  /**
+   * Checks if one of the references contain the project's URL.
+   * 
+   * @param references The list of type {@link ReferenceLink}.
+   * @param project of type {@link GitHubProject}.
+   * @return True if the project's URL is found, false otherwise.
+   */
+  private static boolean match(List<ReferenceLink> references, GitHubProject project) {
+    if (references == null) {
+      return false;
+    }
+
+    for (ReferenceLink reference : references) {
+      Optional<URI> referenceUrl = getUrl(reference.getUrl());
```

Comment:
Could you please add `ReferenceLink.url()` method instead?

Reply:
I did not get this. `ReferenceLink` just has `getUrl()` method. Am I missing something here? Let me know.

Reply:
The `ReferenceLink` can parse and return an URI. In other word, the logic in the static `getUrl()` method should be moved in to the `ReferenceLink`.